### ISSUE TITLE
check_subrc should allow selects at the end of method

### ISSUE THREAD

**Issue by sbcgua** [abaplint/abaplint#3944]
Opinionated, but I think the pattern is real and typical. (did I post it before? hope not)

```abap
method xxx.
  select xyz from txyz ... " or select single
endmethod.
```

Imho the `selects` like above should not trigger the check_subrc. The pattern is that this is a dedictaed selection method, potentially in a DAO class (behind an interface) and thus it only returns selected values which are checked versus `initial` after the call.

Now the rule forces to write useless `if sy-subrc. return ...` and as return is also checked by `unnecessary_return` rule it converts into `if sy-subrc. clear rt_result` which even more ridiculous. So a flag like `allowIfLastStatement` could be nice for such a case.

P.S. Maybe also `read table` is relevant. I don't have an example at hand but some getter method may look like bare `read table`

**Comment by larshp** [comment]
I prefer making rules as simple as possible, without special exceptions

the developer can add the pseudo comment or pragma, clearly signaling to the next developer that it is intentionally ignored

but: adding an option in the rule I think is okay, just default it to most strict/simple

**Comment by sbcgua** [comment]
> Pragmas ...

Oh yes ... any near plans? ;)

In addition. Not sure it is easy to implement, but the below is also not an error (I mean the if is not needed):

<img width="258" height="226" alt="Image" src="https://github.com/user-attachments/assets/8c75377f-bc92-48fd-8a77-f8c168b6068a" />
